day1/day1_binsearch.c

```diff
@@ -38,15 +38,24 @@ int main() {
 
     // Allocate initial memory for the two lists
     left_list  = malloc(capacity * sizeof(int));
-    right_list = malloc(capacity * sizeof(int));
 
-    if (!left_list || !right_list) {
+    if (!left_list){
         fprintf(stderr, "Memory allocation failed\n");
         fclose(file);
         ret = 1;
         goto exit_0;
     }
 
+    right_list = malloc(capacity * sizeof(int));
+
+    if (!right_list) {
+
```

Comment:
It's still not time to be annoying, but one day we can talk about these leftover newlines .... lol